### DISCUSSION TITLE
Need argent help with deployment

### DISCUSSION THREAD

**Discussion by kmnahidulislam** [community/community#66808] [Actions]
### Select Topic Area

General

### Body

I'm facing problem with GitHub Repository (mywebsite) .
I have add here a react app . But when I update my file I facing error with workflow (Check Format workflow run failed for main branch) .
Every time when I update file this problem occured. That's why i am facing problem with automatically deploying my website. Please fix it ...you can understand how this is important for me..I have attached here some photos... please fix it

![Screenshot_2023-09-12-01-49-05-983_com.android.chrome.jpg](https://github.zendesk.com/attachments/token/ZM78t7XMUlCrwqGkETAKrL9FS/?name=Screenshot_2023-09-12-01-49-05-983_com.android.chrome.jpg)
![Screenshot_2023-09-12-01-49-19-241_com.android.chrome.jpg](https://github.zendesk.com/attachments/token/GAsgkp0nVWb5gxQ7dBCcqzwo5/?name=Screenshot_2023-09-12-01-49-19-241_com.android.chrome.jpg)

Have a great day
-Nahidul

**Comment by seyLu** [comment]
Hi @kmnahidulislam,

Can you include the logs next time please to provide more context for other devs to be able to help:
- Go to Actions tab
- Select the failed workflow
- Click the error and paste the error message here
![image](https://github.com/community/community/assets/98249191/816e7513-c4bf-4465-9f2e-2fcaf9e3c3ac)
![image](https://github.com/community/community/assets/98249191/75e71c71-1b24-4b13-8725-800278dffe4d)
![image](https://github.com/community/community/assets/98249191/b9dbb3ac-5ae5-482a-92cc-f2e53526bc07)

It would also help if you include the contents of `.github/workflow/<>.yaml` file. 

Devs wouldve done this if the repo is public, but it's private, I mean ..., so yeah.

**Reply by kmnahidulislam** [reply]
![Screenshot_2023-09-12-17-11-40-938_com.android.chrome.jpg](https://github.com/community/community/assets/144264609/757e26a4-8121-4337-b38a-9e25a786afa0)

![Screenshot_2023-09-12-17-11-55-559_com.android.chrome.jpg](https://github.com/community/community/assets/144264609/098b1f6c-05c0-4e11-ae5d-b7d88edcf9bd)

![Screenshot_2023-09-12-17-11-50-921_com.android.chrome.jpg](https://github.com/community/community/assets/144264609/d2eee122-e787-4dc2-a299-d92ccbd9f715)



Look

**Reply by seyLu** [reply]
In your prettier.yml file, you prolly have:

```diff
run: |
-  npm run check-format
-  prettier -c "./**/*"
+  prettier -c "./**/*"
+  npm run check-format
```

just swap the two commands and it should work.

**Reply by kmnahidulislam** [reply]
sorry to say.. i cannot find the file ....kindly help me to find it . my full file in github (https://github.com/kmnahidulislam/mywebsite)

**Reply by seyLu** [reply]
Hmm, I was just guessing since I didn't have access to the repo in question.

Now that I am seeing it, you have a `prettier.yml` file that runs `npm run check-format`:

![image](https://github.com/community/community/assets/98249191/88b6d684-f888-4a0e-9eec-7ab2eadb8002)

<br>

But if we look at `package.json`, `check-format` script runs `prettier` with the `-c` flag, which doesn't format the files, but only checks if it is properly formatted. 

![image](https://github.com/community/community/assets/98249191/27f0bb96-4156-4c8f-a532-a79ba8d344c1)

To fix this, in your terminal/command line, just run:

```bash
npm run format
```

That should run the prettier script to format your files, such that when the prettier.yml workflow runs again, it shouldn't throw an error.

**Reply by kmnahidulislam** [reply]
as a beginner, I cannot understand properly

![Screenshot 2023-09-12 224305](https://github.com/community/community/assets/144264609/b2737555-4bda-411a-8ba3-71d6ca128364)


kindly look the new error

**Reply by seyLu** [reply]
There were no errors, only warnings. After running `npm run format`, add > commit > push as usual.

**Reply by kmnahidulislam** [reply]
can i fix the npm  vulnerabilities warnings?

**Reply by seyLu** [reply]
If it is a warning, we ignore it. Once you get a bit more experience with javascript and the node ecosystem, you'll know that however much fixing you do, those warnings never go away.

**Reply by kmnahidulislam** [reply]
you mean the warning is normal?

**Reply by seyLu** [reply]
Yup.

**Reply by kmnahidulislam** [reply]
which is better for deploying react app render/vercel? and why?

**Reply by seyLu** [reply]
If it is a static site, why not just use Github Pages? 

But to answer your question, react is a library that a lot of static site generators use/support, eg. [Astro](https://astro.build/). Vercel would make more sense if it's just the frontend.

Render is overkill. Render is great for hosting full stack/backend/apis.

**Reply by kmnahidulislam** [reply]
i have deleted the .github file ... is it required? as a beginner ,i donot know that..after delete it , i'm not facing the workflow error

![Screenshot 2023-09-12 230028](https://github.com/community/community/assets/144264609/652e76ec-35d1-4ff9-a503-1493eda83838)

**Reply by seyLu** [reply]
.github folder is where you put the automated workflows. Like let's say something change in the Github repo, you push the code, and automated tasks run like linters, tests, security scanners, build & deploy, etc. 

As a beginner, I get that it could get a bit overwhelming. It's cool though, you can just manually deploy it using the terminal/command line.

**Reply by kmnahidulislam** [reply]
if i again add .github folder , when i update file in github , it autometically make auto deployment? please donot mind ..i'm beginner

**Reply by seyLu** [reply]
Yup. As long as you have a working deploy and build workflow under .github/workflows dir.
![image](https://github.com/community/community/assets/98249191/4cc40dcd-8c61-4907-aaea-239fa10f6ec5)

**Reply by seyLu** [reply]
If you are interested in learning how to write your own workflows, you can check it out on [Github Actions](https://github.com/features/actions). But again, what you can do automatically, there's nothing wrong doing it manually.

**Reply by kmnahidulislam** [reply]
i am going to add this folder again ,please tell me, which code from .github folder , i need to change?

**Reply by kmnahidulislam** [reply]
and if i add this file directly ,will it shows the workflow error again?

**Reply by seyLu** [reply]
Just run `npm run format` before you add > commit > push, then you shouldn't see the same workflow error.

**Reply by kmnahidulislam** [reply]
where and how to run please explan?

**Reply by seyLu** [reply]
You run it on the terminal/command line `npm run format`.

If you encounter any more errors, consult the gtihub page of the portfolio template [developerFolio](https://github.com/saadpasta/developerFolio). Instructions are pretty clear there, just follow that.

I have to sleep bro, I am sorry, I have an interview tom.

**Reply by kmnahidulislam** [reply]
okay , thanks mam..


after run it ...it shows me:


`NAHIDUL@LENOVO MINGW64 /d/developerFolio-master/developerFolio-master (main)
$ npm run format

> developerfolio@0.1.0 format
> prettier --write "./**/*.{js,css,json,scss}"

blue-belt\.vscode\extensions.json 71ms
blue-belt\.vscode\launch.json 8ms
blue-belt\package.json 4ms
blue-belt\tsconfig.json 3ms
fetch.js 70ms
package.json 8ms
public\manifest.json 6ms
src\_globalColor.scss 81ms
src\App.js 8ms
src\App.scss 14ms
src\App.test.js 23ms
src\components\achievementCard\AchievementCard.js 20ms
src\components\achievementCard\AchievementCard.scss 18ms
src\components\blogCard\BlogCard.js 17ms
src\components\blogCard\BlogCard.scss 42ms
src\components\button\Button.js 7ms
src\components\button\Button.scss 11ms
src\components\displayLottie\DisplayLottie.js 11ms
src\components\educationCard\EducationCard.js 32ms
src\components\educationCard\EducationCard.scss 23ms
src\components\experienceCard\ExperienceCard.js 20ms
src\components\experienceCard\ExperienceCard.scss 25ms
src\components\footer\Footer.js 15ms
src\components\footer\Footer.scss 2ms
src\components\githubProfileCard\GithubProfileCard.js 10ms
src\components\githubProfileCard\GithubProfileCard.scss 10ms
src\components\githubRepoCard\GithubRepoCard.js 18ms
src\components\githubRepoCard\GithubRepoCard.scss 11ms
src\components\header\Header.js 16ms
src\components\header\Header.scss 16ms
src\components\socialMedia\SocialMedia.js 19ms
src\components\socialMedia\SocialMedia.scss 9ms
src\components\softwareSkills\SoftwareSkill.js 5ms
src\components\softwareSkills\SoftwareSkill.scss 5ms
src\components\talkCard\TalkCard.js 5ms
src\components\talkCard\TalkCard.scss 13ms
src\components\ToggleSwitch\ToggleSwitch.js 6ms
src\components\ToggleSwitch\ToggleSwitch.scss 9ms
src\containers\achievement\Achievement.js 5ms
src\containers\achievement\Achievement.scss 5ms
src\containers\blogs\Blog.scss 7ms
src\containers\blogs\Blogs.js 13ms
src\containers\contact\Contact.js 9ms
src\containers\contact\Contact.scss 8ms
src\containers\education\Education.js 3ms
src\containers\education\Education.scss 4ms
src\containers\greeting\Greeting.js 5ms
src\containers\greeting\Greeting.scss 8ms
src\containers\loading\Loading.js 5ms
src\containers\loading\loading.scss 20ms
src\containers\Main.js 9ms
src\containers\Main.scss 2ms
src\containers\podcast\Podcast.js 7ms
src\containers\podcast\Podcast.scss 4ms
src\containers\profile\Profile.js 6ms
src\containers\projects\Project.scss 4ms
src\containers\projects\Projects.js 7ms
src\containers\skillProgress\Progress.scss 5ms
src\containers\skillProgress\skillProgress.js 6ms
src\containers\skills\Skills.js 5ms
src\containers\skills\Skills.scss 4ms
src\containers\splashScreen\SplashScreen.css 7ms
src\containers\splashScreen\SplashScreen.js 5ms
src\containers\StartupProjects\StartupProject.js 6ms
src\containers\StartupProjects\StartupProjects.scss 8ms
src\containers\talks\Talks.js 7ms
src\containers\talks\Talks.scss 4ms
src\containers\topbutton\Top.js 6ms
src\containers\topbutton\Top.scss 6ms
src\containers\twitter-embed\twitter.js 4ms
src\containers\twitter-embed\twitter.scss 3ms
src\containers\workExperience\WorkExperience.js 5ms
src\containers\workExperience\WorkExperience.scss 4ms
src\contexts\StyleContext.js 3ms
src\hooks\useLocalStorage.js 6ms
src\index.css 5ms
src\index.js 2ms
src\portfolio.js 27ms
src\setupTests.js 2ms
src\utils.js 2ms

NAHIDUL@LENOVO MINGW64 /d/developerFolio-master/developerFolio-master (main)
$`
